File: src/main/java/com/clinica/atendimento/dto/ProfissionalDTO.java
package com.clinica.atendimento.dto;

import com.clinica.atendimento.model.Pessoa;
import com.clinica.atendimento.model.Profissional;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;
import com.fasterxml.jackson.annotation.JsonProperty;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Accessors(fluent = true)
@Builder
public class ProfissionalDTO {

    @JsonProperty
    private Long id;
    @JsonProperty
    private PessoaDTO pessoa;

    // Static method to convert from entity to DTO
    public static ProfissionalDTO fromEntity(Profissional profissional) {
        if (profissional == null) {
            return null;
        }

        return ProfissionalDTO.builder()
                .id(profissional.pessoa().id())
                .pessoa(PessoaDTO.fromEntity(profissional.pessoa()))
                .build();
    }

    // Method to convert from DTO to entity
    public Profissional toEntity() {
        Pessoa pessoaEntity = pessoa != null ? pessoa.toEntity() : null;

        // Set the ID on the Pessoa entity
        if (pessoaEntity != null && id != null) {
            pessoaEntity.id(id);
        }

        return Profissional.builder()
                .pessoa(pessoaEntity)
                .build();
    }
}

File: src/main/kotlin/com/clinica/atendimento/dto/ProfissionalDTO.kt
package com.clinica.atendimento.dto

import com.clinica.atendimento.model.Pessoa
import com.clinica.atendimento.model.Profissional
import com.fasterxml.jackson.annotation.JsonProperty

data class ProfissionalDTO(
    @JsonProperty
    var id: Long? = null,
    
    @JsonProperty
    var pessoa: PessoaDTO? = null
) {
    // Fluent accessors to maintain compatibility with Java code
    fun id(): Long? = id
    
    fun id(id: Long?): ProfissionalDTO {
        this.id = id
        return this
    }
    
    fun pessoa(): PessoaDTO? = pessoa
    
    fun pessoa(pessoa: PessoaDTO?): ProfissionalDTO {
        this.pessoa = pessoa
        return this
    }

    // Method to convert from DTO to entity
    fun toEntity(): Profissional {
        val pessoaEntity = pessoa?.toEntity()

        // Set the ID on the Pessoa entity
        if (pessoaEntity != null && id != null) {
            pessoaEntity.id(id)
        }

        return Profissional.builder()
            .pessoa(pessoaEntity)
            .build()
    }

    companion object {
        // Static method to convert from entity to DTO
        @JvmStatic
        fun fromEntity(profissional: Profissional?): ProfissionalDTO? {
            if (profissional == null) {
                return null
            }

            return builder()
                .id(profissional.pessoa().id())
                .pessoa(PessoaDTO.fromEntity(profissional.pessoa()))
                .build()
        }

        @JvmStatic
        fun builder(): ProfissionalDTOBuilder {
            return ProfissionalDTOBuilder()
        }
    }

    class ProfissionalDTOBuilder {
        private val instance = ProfissionalDTO()

        fun id(id: Long?): ProfissionalDTOBuilder {
            instance.id = id
            return this
        }

        fun pessoa(pessoa: PessoaDTO?): ProfissionalDTOBuilder {
            instance.pessoa = pessoa
            return this
        }

        fun build(): ProfissionalDTO {
            return instance
        }
    }
}